src/rpc/evo.cpp

```diff
@@ -1267,25 +1267,39 @@ static bool CheckWalletOwnsScript(CWallet* pwallet, const CScript& script) {
 }
 #endif
 
-static UniValue BuildDMNListEntry(CWallet* pwallet, const CDeterministicMN& dmn, bool detailed)
+static UniValue BuildDMNListEntry(CWallet* pwallet, const CDeterministicMN& dmn, bool detailed, const CBlockIndex* pindex = nullptr)
 {
     if (!detailed) {
         return dmn.proTxHash.ToString();
     }
 
     UniValue o = dmn.ToJson();
 
+    CTransactionRef collateralTx{nullptr};
     int confirmations = GetUTXOConfirmations(dmn.collateralOutpoint);
+
+    if (pindex != nullptr) {
+        if (confirmations > -1) {
+            confirmations -= WITH_LOCK(cs_main, return ::ChainActive().Height()) - pindex->nHeight;
+        } else {
+            uint256 minedBlockHash;
+            collateralTx = GetTransaction(/* pindex */ nullptr, /* mempool */ nullptr, dmn.collateralOutpoint.hash, Params().GetConsensus(), minedBlockHash);
+            const CBlockIndex* const pindexMined = WITH_LOCK(cs_main, return g_chainman.m_blockman.LookupBlockIndex(minedBlockHash));
+            CHECK_NONFATAL(pindexMined != nullptr);
+            CHECK_NONFATAL(pindex->nHeight >= pindexMined->nHeight);
+            confirmations = pindex->nHeight - pindexMined->nHeight + 1;
```

Comment:
should be there also check through `pindex->GetAncestor(...)` to be sure that are same fork?